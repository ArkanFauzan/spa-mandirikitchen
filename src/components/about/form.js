import React from "react"

export function FormAbout(props){

    const _handleChange = (event) => {
        props.onChangeValue(event.target)
    }

    const _handleSubmit = (event) => {
        event.preventDefault();
        const {nama, email, pesan} = props.state
        alert(`Nama: ${nama}, Email: ${email}, Pesan: ${pesan}`)
    }

    const {nama,email,pesan} = props.state;

    return(
        <form onSubmit={_handleSubmit} style={{display:"flex",flexDirection:"column",maxWidth:"600px"}}>
                    
                <label>Nama</label>
                <input type="text" name="nama" value={nama} onChange={_handleChange} />
                    
                <label>Email</label>
                <input type="text" name="email" value={email} onChange={_handleChange}  />
                    
                <label>Pesan</label>
                <textarea type="text" name="pesan" value={pesan} onChange={_handleChange}  />
            <button className="my-4 btn btn-primary" type="submit">Kirim</button>
        </form>
    )
}