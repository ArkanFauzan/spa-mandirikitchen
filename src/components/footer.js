import React from "react"

export class FooterComponent extends React.Component{
    render(){
        return(
            <>
                <div>
                    <br/>
                    <h3>Hubungi kami</h3>
                    0812 xxxx xxxx (A.N. Gunadi) <br/>0821 xxxx xxxx (A.N. Heri)
                </div><br/>
                <div style={{borderTop:"solid 1px black"}}>
                    <br/>
                    Mandiri Kitchen 2020 all rights reserved
                </div><br/>
            </>
        )
    }
}