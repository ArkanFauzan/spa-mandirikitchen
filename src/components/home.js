import React from "react"
import { Alert, Card, Button
} from 'react-bootstrap';

import {Link, BrowserRouter as Router} from 'react-router-dom'
import {listService} from "./home/service"
import {JumbotronComponenet} from "./home/jumbotron"

export class HomeComponent extends React.Component{
    render(){
        return(
            <Router>
            {/* jumbotron */}
            <JumbotronComponenet />
            
            {/* Card Our service */}
            <div className="row pt-2 bg-light" style={{textAlign:'center'}}>
                {/* menampilkan list service */}
                {listService}

                <div className="my-4 col-12">
                    <Button variant="primary" size="lg" style={{margin:'auto'}}><Link style={{color:'white',textDecoration:'none'}} to="products">Let's See Our Product -&gt;</Link></Button>
                </div>
            </div>

            </Router>
        )
    }
}