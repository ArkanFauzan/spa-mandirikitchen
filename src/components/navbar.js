import React from 'react';
import { Navbar, NavDropdown, Button,
    Nav, Form, FormControl
} from 'react-bootstrap';

import {BrowserRouter as Router,
    Switch,
    Route,
    Link
} from 'react-router-dom'

export function NavbarComponent(){
    return(
        <Navbar fixed="top" bg="light" expand="lg">
            <Navbar.Brand ><Link to="/home"><img style={{marginRight:"50px",}} width="150px" src="img/logo-crop.jpg"/></Link></Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
              <Nav className="mr-auto">
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/home">HOME</Link></Nav.Link>
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/products">PRODUCTS</Link></Nav.Link>
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/about">ABOUT</Link></Nav.Link>
              </Nav>
              <Form inline>
                <FormControl type="text" placeholder="Search" className="mr-sm-2" />
                <Button variant="outline-success">Search</Button>
              </Form>
            </Navbar.Collapse>
          </Navbar>
    )
}