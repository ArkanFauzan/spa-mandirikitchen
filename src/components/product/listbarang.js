import React from "react"

export class ListBarang extends React.Component{
    render(){
        const {barang,filter} = this.props;

        // melakukan filter barang
        let barangFilter = barang;
        if(filter!=="semua kategori"){
            barangFilter = barang.filter(value=>{
                return value.kategori===filter
            })
        }

        // membuat list barang berdasarkan barang yang telah difilters
        const list = barangFilter.map((value,index)=>{
            return(
                <div style={{margin:"30px 0px",padding:"0px 10px"}} key={index}>
                    <h3>{index+1}. {value.nama}</h3>
                    <div style={{display:"flex",flexDirection:"column",paddingRight:"20px"}}>
                        <img style={{display:"block",margin:"20px auto", width:"auto"}} src={value.img} />
                        <h4 style={{display:"block",textAlign:"center"}}>{value.deskripsi}</h4>
                        <img style={{display:"block",margin:"20px auto"}} src={value.harga} />
                    </div>
                </div>
            )
        });

        console.log(barangFilter);

        return(
            <>{list}</>
        )
    }
}