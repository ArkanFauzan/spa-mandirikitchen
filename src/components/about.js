import React from "react"
import {FormAbout} from "./about/form"

export class AboutComponent extends React.Component{
    constructor(){
        super()
        this.state={
            nama:"",
            email:"",
            pesan:"",
        }
    }

    _handleChange = (value) => {
        this.setState({[value.name]:value.value})
    }

    render(){

        const {nama, email, pesan} = this.state

        return(
            <div className="container">
                <img style={{display:"block", margin:"auto"}} width="100%"  src="img/cv-crop.jpg" />
                <h2 className="my-4">CV Mandiri Kitchen Perkasa</h2>
                <p className="mb-4" style={{fontSize:"20px",textAlign:"justify"}}>
                    Mandiri Kitchen Perkasa adalah sebuah perusahaan yang bergerak dibidang 
                    disain dapur restoran serta pembuatan peralatan dapur komersial. Disain 
                    atau perencanaan dapur restoran sangat diperlukan agar ruang dapur 
                    restoran anda menjadi efisien dan fungsional serta biaya yang dikeluarkan 
                    sesuai dengan hasil yg diinginkan karena Penataan Peralatan Memasak yang 
                    TEPAT dan BENAR.  Ruang lingkup dan spesifikasi pekerjaan kami 
                    meliputi Desain dan perencanaan dapur komersial, Pembuatan dan pengadaan 
                    alat alat dapur komersial, instalasi ducting dan instalasi gas restoran. 
                    Yang dimaksud Dapur Komersial disini adalah dapur yang dipakai untuk 
                    keperluan usaha, seperti Dapur Restoran, Dapur Hotel, Dapur Asrama maupun 
                    Dapur Catering</p>

                <h2>Kontak kami</h2>
                <table className="my-4">
                    
                    <tr>
                        <td>Gunadi</td>
                        <td>&nbsp;&nbsp;&nbsp;: 0821 xxxx xxxx</td>
                    </tr>
                    <tr>
                        <td>Heri</td>
                        <td>&nbsp;&nbsp;&nbsp;: 0812 xxxx xxxx</td>
                    </tr>
                    <tr>
                        <td>Website</td>
                        <td>&nbsp;&nbsp;&nbsp;: mandirikitchenperkasa.com</td>
                    </tr>
                    <tr>
                        <td>Email</td>
                        <td>&nbsp;&nbsp;&nbsp;: mandirikitchenperkasa@gmail.com</td>
                    </tr>
                    <tr>
                        <td>Alamat</td>
                        <td>&nbsp;&nbsp;&nbsp;: Depok, Jawa Barat</td>
                    </tr>
                </table>

                <h2>Atau melalui form berikut:</h2>

                {/* menampilkan controlled form dengan prinsip lifting state up */}
                <FormAbout state={this.state} onChangeValue={this._handleChange}/>
            </div>
        )
    }
}