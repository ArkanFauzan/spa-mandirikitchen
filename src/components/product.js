import React,{useState, useEffect, useContext} from "react" 
import {BarangContext} from "../barangContext"

// import list barang
import {ListBarang} from "./product/listbarang"

export function ProductComponent(){

    // menggunakan hooks untuk memfilter kategori
    const [state, Filter] = useState({filter:"semua kategori"}) ;
    const {filter} = state;

    // mengambil nilai barang dari provider (menggunakan Context API)
    const {barang} = useContext(BarangContext);

    // mengubah state filter
    const _handleChange = (event) => {
        Filter({filter:event.target.value})
    }

    // similar componentDidUpdate(){}
    useEffect(()=>{
        alert(`Produk yang ditampilkan berdasarkan kategori ${filter}`)
    },[filter])

    return(
        <div className="container-fluid" style={{marginTop:"100px"}}>
            <form>
                <label><p style={{fontSize:"25px",textAlign:"justify"}}>Tampilkan berdasarkan : </p></label>
                <select  style={{fontSize:"25px",padding:"7px 15px"}} name="filter" value={filter} onChange={_handleChange}>
                    <option value="semua kategori">Semua Kategori</option>
                    <option value="area memasak">Area Memasak</option>
                    <option value="area mencuci">Area Mencuci</option>
                </select>
            </form>

            {/* menampilkan list barang sesuai filter kategori */}
            <ListBarang barang={barang} filter={filter}/>
        </div>
    )
}