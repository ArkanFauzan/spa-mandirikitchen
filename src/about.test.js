import React from 'react';
import {ProductComponent} from './components/product';
import {AboutComponent} from './components/about';
import renderer from 'react-test-renderer';

it('About is renders correctly', () => {
    let component = renderer.create(<AboutComponent/>);  
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    tree.props._handleChange;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });
