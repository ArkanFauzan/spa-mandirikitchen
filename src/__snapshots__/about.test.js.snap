// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About is renders correctly 1`] = `
<div
  className="container"
>
  <img
    src="img/cv-crop.jpg"
    style={
      Object {
        "display": "block",
        "margin": "auto",
      }
    }
    width="100%"
  />
  <h2
    className="my-4"
  >
    CV Mandiri Kitchen Perkasa
  </h2>
  <p
    className="mb-4"
    style={
      Object {
        "fontSize": "20px",
        "textAlign": "justify",
      }
    }
  >
    Mandiri Kitchen Perkasa adalah sebuah perusahaan yang bergerak dibidang disain dapur restoran serta pembuatan peralatan dapur komersial. Disain atau perencanaan dapur restoran sangat diperlukan agar ruang dapur restoran anda menjadi efisien dan fungsional serta biaya yang dikeluarkan sesuai dengan hasil yg diinginkan karena Penataan Peralatan Memasak yang TEPAT dan BENAR.  Ruang lingkup dan spesifikasi pekerjaan kami meliputi Desain dan perencanaan dapur komersial, Pembuatan dan pengadaan alat alat dapur komersial, instalasi ducting dan instalasi gas restoran. Yang dimaksud Dapur Komersial disini adalah dapur yang dipakai untuk keperluan usaha, seperti Dapur Restoran, Dapur Hotel, Dapur Asrama maupun Dapur Catering
  </p>
  <h2>
    Kontak kami
  </h2>
  <table
    className="my-4"
  >
    <tr>
      <td>
        Gunadi
      </td>
      <td>
           : 0821 xxxx xxxx
      </td>
    </tr>
    <tr>
      <td>
        Heri
      </td>
      <td>
           : 0812 xxxx xxxx
      </td>
    </tr>
    <tr>
      <td>
        Website
      </td>
      <td>
           : mandirikitchenperkasa.com
      </td>
    </tr>
    <tr>
      <td>
        Email
      </td>
      <td>
           : mandirikitchenperkasa@gmail.com
      </td>
    </tr>
    <tr>
      <td>
        Alamat
      </td>
      <td>
           : Depok, Jawa Barat
      </td>
    </tr>
  </table>
  <h2>
    Atau melalui form berikut:
  </h2>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "maxWidth": "600px",
      }
    }
  >
    <label>
      Nama
    </label>
    <input
      name="nama"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      Email
    </label>
    <input
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      Pesan
    </label>
    <textarea
      name="pesan"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="my-4 btn btn-primary"
      type="submit"
    >
      Kirim
    </button>
  </form>
</div>
`;

exports[`About is renders correctly 2`] = `
<div
  className="container"
>
  <img
    src="img/cv-crop.jpg"
    style={
      Object {
        "display": "block",
        "margin": "auto",
      }
    }
    width="100%"
  />
  <h2
    className="my-4"
  >
    CV Mandiri Kitchen Perkasa
  </h2>
  <p
    className="mb-4"
    style={
      Object {
        "fontSize": "20px",
        "textAlign": "justify",
      }
    }
  >
    Mandiri Kitchen Perkasa adalah sebuah perusahaan yang bergerak dibidang disain dapur restoran serta pembuatan peralatan dapur komersial. Disain atau perencanaan dapur restoran sangat diperlukan agar ruang dapur restoran anda menjadi efisien dan fungsional serta biaya yang dikeluarkan sesuai dengan hasil yg diinginkan karena Penataan Peralatan Memasak yang TEPAT dan BENAR.  Ruang lingkup dan spesifikasi pekerjaan kami meliputi Desain dan perencanaan dapur komersial, Pembuatan dan pengadaan alat alat dapur komersial, instalasi ducting dan instalasi gas restoran. Yang dimaksud Dapur Komersial disini adalah dapur yang dipakai untuk keperluan usaha, seperti Dapur Restoran, Dapur Hotel, Dapur Asrama maupun Dapur Catering
  </p>
  <h2>
    Kontak kami
  </h2>
  <table
    className="my-4"
  >
    <tr>
      <td>
        Gunadi
      </td>
      <td>
           : 0821 xxxx xxxx
      </td>
    </tr>
    <tr>
      <td>
        Heri
      </td>
      <td>
           : 0812 xxxx xxxx
      </td>
    </tr>
    <tr>
      <td>
        Website
      </td>
      <td>
           : mandirikitchenperkasa.com
      </td>
    </tr>
    <tr>
      <td>
        Email
      </td>
      <td>
           : mandirikitchenperkasa@gmail.com
      </td>
    </tr>
    <tr>
      <td>
        Alamat
      </td>
      <td>
           : Depok, Jawa Barat
      </td>
    </tr>
  </table>
  <h2>
    Atau melalui form berikut:
  </h2>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "maxWidth": "600px",
      }
    }
  >
    <label>
      Nama
    </label>
    <input
      name="nama"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      Email
    </label>
    <input
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      Pesan
    </label>
    <textarea
      name="pesan"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="my-4 btn btn-primary"
      type="submit"
    >
      Kirim
    </button>
  </form>
</div>
`;
